tests/integration/security/authorization_test.go

```diff
@@ -638,296 +637,366 @@ func TestAuthorization_EgressGateway(t *testing.T) {
+			b := apps.B.Match(echo.Namespace(apps.Namespace1.Name()))
+			c := apps.C.Match(echo.Namespace(apps.Namespace2.Name()))
+
+			newTestCase := func(name string, from, target echo.Instances, port string, expectAllowed bool, scheme scheme.Instance) rbacUtil.TestCase {
+				return rbacUtil.TestCase{
+					NamePrefix: name,
+					Request: connection.Checker{
+						From: from[0],
+						Options: echo.CallOptions{
+							Target:   target[0],
+							PortName: port,
+							Scheme:   scheme,
+							Path:     "/data",
+						},
+					},
+					ExpectAllowed: expectAllowed,
+				}
 			}
-			policies := tmpl.EvaluateAllOrFail(t, args,
-				file.AsStringOrFail(t, "testdata/authz/v1beta1-egress-gateway.yaml.tmpl"))
-			t.Config().ApplyYAMLOrFail(t, "", policies...)
 
 			cases := []struct {
-				name  string
-				path  string
-				code  string
-				body  string
-				host  string
-				from  echo.Workload
-				token string
+				configFile string
+				configSrc  string
+				configDst  string
+				subcases   []rbacUtil.TestCase
 			}{
 				{
-					name: "allow path to company.com",
-					path: "/allow",
-					code: response.StatusCodeOK,
-					body: "handled-by-egress-gateway",
-					host: "www.company.com",
-					from: getWorkload(a, t),
-				},
-				{
-					name: "deny path to company.com",
-					path: "/deny",
-					code: response.StatusCodeForbidden,
-					body: "RBAC: access denied",
-					host: "www.company.com",
-					from: getWorkload(a, t),
-				},
-				{
-					name: "allow service account a to a-only.com over mTLS",
-					path: "/",
-					code: response.StatusCodeOK,
-					body: "handled-by-egress-gateway",
-					host: "a-only.com",
-					from: getWorkload(a, t),
+					configFile: "testdata/authz/v1beta1-tcp-1.yaml.tmpl",
+					configDst:  util.BSvc,
+					subcases: []rbacUtil.TestCase{
+						// The policy on workload b denies request with path "/data" to port 8090:
+						// - request to port http-8091 should be denied because both path and port are matched.
+						// - request to port http-8092 should be allowed because the port is not matched.
+						// - request to port tcp-8093 should be allowed because the port is not matched.
+						newTestCase("port and path/", a, b, "http-8091", false, scheme.HTTP),
+						newTestCase("port and path/", a, b, "http-8092", true, scheme.HTTP),
+						newTestCase("port and path/", a, b, "tcp-8093", true, scheme.TCP),
+						newTestCase("port and path/", vm, b, "http-8091", false, scheme.HTTP),
+						newTestCase("port and path/", vm, b, "http-8092", true, scheme.HTTP),
+						newTestCase("port and path/", vm, b, "tcp-8093", true, scheme.TCP),
+					},
 				},
 				{
-					name: "deny service account c to a-only.com over mTLS",
-					path: "/",
-					code: response.StatusCodeForbidden,
-					body: "RBAC: access denied",
-					host: "a-only.com",
-					from: getWorkload(c, t),
+					configFile: "testdata/authz/v1beta1-tcp-1.yaml.tmpl",
+					configDst:  util.VMSvc,
```

Comment:
are't we making the test workload agnostic? why are we still duplicating the test cases for b and vm here.